pandas/plotting/_matplotlib/core.py

```diff
@@ -489,6 +489,18 @@ def _post_plot_logic(self, ax, data):
         """Post process for each axes. Overridden in child classes"""
         pass
 
+    def _has_iterable_only_float_int_or_datetime(self, list_iterable: list) -> bool:
+        """checks if an iterable has float,int or datetime datatype """
+        """targeting #GH40781"""
+        from datetime import datetime as dt
+        if (
+            all(([isinstance(x, dt) for x in list_iterable]))
```

Comment:
use is_* accessors

Reply:
I tried implementing `is_datetime*` , `is_datetime_or_timedelta*` etc and could not get it to work with raw `datetime.datetime` formats. Any suggestions?

Reply:
I think there are existing implementations, so should be no need to implement new ones

Reply:
I mean I tried to use those existing implementations, but could not get it to work with `datetime` objects being passed as `xlim` or `ylim`

Reply:
emm, you meant non of those implementations of is_* accessors in `https://github.com/pandas-dev/pandas/blob/master/pandas/core/dtypes/common.py` could work with `datetime` objects? That sounds a bit weird, could you pls elaborate or maybe post a new issue to address this?

Reply:
I will try more samples and recreate an issue if that verifies with  exact example soon!